src/electron/background.ts

```diff
@@ -56,21 +57,36 @@ import "@/ipc/vector-store";
 // Import keys
 import "@/ipc/keys";
 
+// Obtain a lock to check if the app is locked (primary instance)
+// If false, we close the app, as we want to prevent multiple instances of our app
+// https://github.com/electron/electron/blob/v30.0.0-nightly.20240221/docs/api/app.md#apprequestsingleinstancelockadditionaldata
+const gotTheLock = app.requestSingleInstanceLock();
+
 let unsubscribe: (() => Promise<void>) | undefined;
-// Initialize the application by calling the main function.
-// Upon completion, log to the console indicating the application has started.
-main().then(() => {
-	logger.info("Application started successfully");
-	unsubscribe = watchStores();
-});
-
-// Listen for the 'window-all-closed' event on the Electron app object.
-// This event is emitted when all windows of the application have been closed.
-// In response, quit the application to free up resources, adhering to typical desktop application
-// behavior.
-app.on("window-all-closed", () => {
+
+if (gotTheLock || isTest) {
+	// Initialize the application by calling the main function.
+	// Upon completion, log to the console indicating the application has started.
```

Comment:
```suggestion
```

Reply:
Comment seems useless, as its clear that something is logged to the console.

Reply:
this was not a change of the PR though. 

But I think the flag here is not actually needed anymore as it was just aiming to fix the broken tests. I later identified the actual root cause.

I will remove the flag condition and remove the comment along the way.

Reply:
seems like the flag was actually needed afterall. Comments were removed